### ISSUE TITLE
Sushid strange behaviour and spamming of transaction messages

### ISSUE THREAD

**Issue by kingsleyh** [SushiChain/SushiChain#205]
## Checklist
- [x] You've tried latest version of SushiChain
- [x] You've confirmed there is not a same report

## Description
When I start a local sushid and then make a transaction - the local sushid constantly spams a message saying `new transaction coming` and then when I start the miner it constantly spams out the message that the transaction is invalid once the first block is processed. And then trying to make another transaction just hangs. 

## Environment

- Using docker: no
- OSX
- Crystal version: 0.24.2
- Where do you connect to? http://localhost:3000

## Command Lines and results

1. start a local sushid from master branch
```bash
> sushid -w w1.json -n http://localhost:3000 --testnet --private 
```

2. make an invalid transaction (or any transaction I think)
```bash
> sushi tx create -f 0.0001 -m 1 -a VDBkYWQxZjZlZjllOTAzYzNiODQ0NmZkZTI4NDBhYmMzYjUxYThjM2E1ZjNkODlj -w w1.json -n http://localhost:3000 
```

3. look at the sushid process logging into the console - it's spamming out:

```
w transaction coming: 115e84b9354a140345e3d80170acd54b274a681793d1483d518447f41b599cd8
[ 2018-05-28 21:16:59 --    Info ] new transaction coming: 115e84b9354a140345e3d80170acd54b274a681793d1483d518447f41b599cd8
```

**Comment by kingsleyh** [comment]
Ok I have done some more investigation.

The last commit where everything works fine before this problem occurs is: 
`88965e3801dd84286badfac6dea8d53d3d1dd516` 

The commit that introduced the problem seems to be this one: `97419c690da6df3f2067d1615912d034af6b12c0`

![1__tig](https://user-images.githubusercontent.com/103770/40733427-80f6c20e-642d-11e8-9ccf-766e73e3d5d0.jpg)

The problem is that the message is printed out to the log constantly forever and it blocks other things from happening:

![3__bash_and_2__sushid](https://user-images.githubusercontent.com/103770/40733443-9289ddbc-642d-11e8-8dd2-4fb6a7025ca9.jpg)

**Comment by kingsleyh** [comment]
The problem also happens on Docker running locally - however the problem does not seem to happen on the AWS testnet server.

**Comment by kingsleyh** [comment]
Ok I've found the bug - it's to do with `--private` - the bug does not occur when running a non private node - thats why testnet is working fine. When the `sushid` is started in in private the `send_on_chord` logic get stuck in a loop somehow

**Comment by kingsleyh** [comment]
@tbrand so the commit that broke it - actually fixed a bug related to removing an `is_private` for the first connection node check. But this then exposed this bug - which deals with broadcasting a new transaction on chord. - I don't really understand the send_on_chord logic - it seems not quite right - I think perhaps - the `unless` should be an `if` and the logic further down should be shuffled around?

```
  def send_on_chord(message_type, content, from : Chord::NodeContext? = nil)
      _nodes = @chord.find_nodes

      unless @is_private
        _nodes[:private_nodes].each do |private_node|
          next if !from.nil? && from[:is_private] && private_node[:context][:id] == from[:id]
          send(private_node[:socket], message_type, content)
        end

        if successor = _nodes[:successor]
          if successor[:context][:id] != content[:from][:id]
            send(successor[:socket], message_type, content)
          end
        end
      else
        if from.nil? || from[:is_private]
          if successor = _nodes[:successor]
            send(successor[:socket], message_type, content)
          end
        end
      end
    end
```

**Comment by kingsleyh** [comment]
Fixed by #206 - great work!
